File: Week2/D2.py
n, k = map(int, input().split())

plate = [[0 for _ in range(n + 2)]]
for i in range(n) :
    plate.append([0] + list(map(int, input().split()))[:n] + [0])
plate.append([0 for _ in range(n + 2)])

result = 0
for h in range(1, n + 1) :
    for w in range(1, n + 1) :
        goormCount = 0
        if plate[h][w] == 0 :
            for li in plate[h - 1:h + 2] :
                for px in li[w - 1:w + 2] :
                    if px == 1 : goormCount += 1
        if goormCount == k : result += 1

print(result)



"""
코드 해설

1. 한 변 길이(이하 n), 찾는 값(이하 k) 입력
2. (n + 2) * (n + 2)의 2차원 배열(이하 plate) 생성, 1행(열) 또는 마지막행(열)은 제외하고 값 입력
3. 배열 탐색
    1) 2행 2열 칸부터 시작하여 값 확인(1이면 계산 안 함)
    2) 확인한 값(i행 j열)을 제외하고 plate를 (i - 1)행부터 (i + 1)행, (j - 1)열부터 (j + 1)까지의 값 중 1의 개수 구하여 k와 비교
4. 결과 출력
"""

File: Week2/D2.java
import java.util.*;

class Week2_2 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt(), k = sc.nextInt();

        String[][] plate = new String[n + 2][n + 2];
        for (int i = 0; i < n + 2; i++) {
            plate[0][i] = "0";
            plate[n + 1][i] = "0";
        }
        for (int i = 1; i < n + 1; i++) {
            plate[i][0] = "0";
            plate[i][n + 1] = "0";
            for (int j = 1; j < n + 1; j++) {
                plate[i][j] = sc.next();
            }
        }

        int result = 0;
        for (int h = 1; h < n + 1; h++) {
            for (int w = 1; w < n + 1; w++) {
                int count = 0;
                if (plate[h][w].equals("0")) {
                    for (int sh = h - 1; sh < h + 2; sh++) {
                        for (int sw = w - 1; sw < w + 2; sw++) {
                            if (plate[sh][sw].equals("1"))
                                count++;
                        }
                    }
                }
                if (count == k)
                    result++;
            }
        }

        System.out.println(result);
    }
}



/**
코드 해설

1. 한 변 길이(이하 n), 찾을 깃발 개수(이하 k) 입력
2. (n + 2) * (n + 2) 크기의 2차원 배열(이하 plate) 생성 후 1행(열) 또는 마지막행(열)은 제외하고 값 입력
3. 배열 탐색
	1) 배열에 값을 입력한 칸만 탐색
	2) 해당 칸의 값이 1인 경우 패스, 0이라면 그 칸을 중심으로 3×3 구간 탐색
	3) 1의 개수가 k와 같다면 결과 값에 추가
4. 결과 출력
 */